### PR TITLE
[stable/jenkins] Remove stale ssh key references in JCasC auto-reload

### PR DESCRIPTION
**Pull request by wmcdona89**
#### What this PR does / why we need it:
Update JCasC auto-reload docs and remove stale ssh key references from version "1.8.0 JCasC auto reload works without ssh keys"

Removes:
`master.adminSshKey`
`master.sidecars.configAutoReload.sshTcpPort`
template `jenkins.gen-key`

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by torstenwalter** [comment]
@wmcdona89 Thanks for catching this! Could you re-base your changes on master?

File: stable/jenkins/templates/_helpers.tpl [hunk 1]
```diff
@@ -171,7 +171,7 @@ unclassified:
 {{- end -}}
 
 {{/*
-Generate private key for jenkins CLI
+Generate private key for jenkins CLI (deprecated)
```
Comment (torstenwalter) [inline stable/jenkins/templates/_helpers.tpl]: I think this can even be removed.

Comment (wmcdona89) [reply stable/jenkins/templates/_helpers.tpl]: removed

**Comment by wmcdona89** [comment]
@torstenwalter Rebased and removed template `jenkins.gen-key`. My understanding is the 2nd commit message will be discarded on merge. Let me know if this is not the case.

**Review by torstenwalter — APPROVED** [review]
/lgtm

**Pull request merged**